src/curves/bip32Ed25519.ts

```diff
@@ -33,7 +34,11 @@ export const getPublicKey = async (
   privateKey: Uint8Array,
   _compressed?: boolean,
 ): Promise<Uint8Array> => {
-  return privateKey.slice(0, 32); // FIXME: this is just a placeholder, fixed in later commits
+  await libsodium.ready;
+  return libsodium.crypto_scalarmult_ed25519_base_noclamp(
```

Comment:
I put the changes around public key in separate commit so I can easily replace it once I figure out how to use `noble` here instead of `libsodium`